plugins/woocommerce/includes/admin/class-wc-admin-menus.php

```diff
@@ -168,7 +168,7 @@ public function save_settings() {
 		global $current_tab, $current_section;
 
 		// We should only save on the settings page.
-		if ( ! is_admin() || ! isset( $_GET['page'] ) || 'wc-settings' !== $_GET['page'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( ! is_wc_admin_settings_page() ) {
```

Comment:
Note: same as original, but adds `is_admin()` check, which looks safe based on the usage context.